src/sentry/api/endpoints/auth_index.py

```diff
@@ -35,6 +35,82 @@ class AuthIndexEndpoint(Endpoint):
 
     permission_classes = ()
 
+    @staticmethod
+    def _reauthenticate_with_sso(request, org_id):
+        """
+        If a user without a password is hitting this, it means they need to re-identify with SSO.
+        """
+        redirect = request.META.get("HTTP_REFERER", "")
+        if not is_safe_url(redirect, allowed_hosts=(request.get_host(),)):
+            redirect = None
+
+        initiate_login(request, redirect)
+        raise SsoRequired(Organization.objects.get_from_cache(id=org_id))
+
+    def _verify_user_via_inputs(self, validator, request):
```

Comment:
since the helper methods we added don't use self, you can make it a static method